Improve performance documentation (#10555)
istio/istio.io: content/en/docs/ops/deployment/performance-and-scalability/index.md

### CPU and memory

Since the sidecar proxy performs additional work on the data path, it consumes CPU

> and memory. As of Istio {{< istio_release_name >}}, a proxy consumes about 0.5 vCPU per 1000

Reviewer: ```suggestion
and memory. In Istio {{< istio_release_name >}}, a proxy consumes about 0.5 vCPU per 1000
```

and memory. In Istio {{< istio_release_name >}}, a proxy consumes about 0.5 vCPU per 1000
requests per second.

The memory consumption of the proxy depends on the total configuration state the proxy holds.
A large number of listeners, clusters, and routes can increase memory usage.
Istio 1.1 introduced namespace isolation to limit the scope of the configuration sent
to a proxy. In a large namespace, the proxy consumes approximately 50 MB of memory.
In a large namespace with [namespace isolation](/docs/reference/config/networking/sidecar/) enabled, the proxy consumes approximately 50 MB of memory.

Since the proxy normally doesn't buffer the data passing through,
request rate doesn't affect the memory consumption.

### Latency

Since Istio injects a sidecar proxy on the data path, latency is an important
consideration. Istio adds an authentication filter and a telemetry filter and a metadata exchange filter

> to the proxy. Every additional filter adds to the path length inside the proxy and affects latency.

Reviewer 2: > Every feature Istio adds

This is meant to say "features that add filters to Envoy".  I'm worried people might read it to say "if I just configure an existing feature, or I configure something that is built into Envoy, it will make Istio slower".

> Every additional feature Istio adds also adds to the path length inside the proxy and potentially affects latency.

Reviewer: ```suggestion
Every feature Istio adds also adds to the path length inside the proxy and potentially affects latency.
```

immediately. This process adds to the queue wait time of the next request and affects
average and tail latencies. The actual tail latency depends on the traffic pattern.

### Latency for Istio {{< istio_release_name >}}

Inside the mesh, a request traverses the client-side proxy and then the server-side
proxy. In the default configuration of Istio {{< istio_release_name >}} (i.e. Istio with telemetry v2),
the two proxies add about 1.7 ms and 2.7 ms to the 90th and 99th percentile latency, respectively, over the baseline data plane latency.
We obtained these results using the [Istio benchmarks](https://github.com/istio/tools/tree/{{< source_branch_name >}}/perf/benchmark)
for the `http/1.1` protocol, with a 1 kB payload at 1000 requests per second using 16 client connections, 2 proxy workers and mutual TLS enabled.

{{< image width="90%"
    link="latency_p90_fortio_with_jitter.svg"
    alt="P90 latency vs client connections"
    caption="P90 latency vs client connections"
>}}

{{< image width="90%"
    link="latency_p99_fortio_with_jitter.svg"
    alt="P99 latency vs client connections"

>     caption="P99 latency vs client connections"

Reviewer 3: Does it remove all discussion about jitter?
We should still mention jitter here, so that if someone tries to run benchmarks with other tools, these numbers will look non reproducible.

Author: I did intentionally remove all discussion of jitter because it has been causing confusion (people thinking 'jitter' is an Istio feature, or that benchmarks without jitter are valid).

We do link to the exact scripts we run
